### PR TITLE
feat (cluster): [AKS v1.23] bump aks, kured versions and some minors

### PR DESCRIPTION
**Pull request by ferantivero**
closes: #527157

## WHY?

aks baseline should remain up to the latest aks version being released.

## WHAT Changed?

- bump aks cluster version up to `1.23`
- bump kured up to `1.9.2`
- bicep cluster-stamp
   - now all rbac roles are `existing` resources
   - remove `reference` function usage in favor of using the actual resource
- add `eastus2` location to acr and cluster-stamp being a friendly reminder for those who need to deploy to another region without issues.

## HOW to Test?

you should be able to clone and follow the instructions to provision a new AKS cluster in version `1.23`. Cluster looks healthy and you can execute the e2e validation without issues.

### CODE REVIEW

File: 05-bootstrap-prep.md
Comment (ckittel) [inline 05-bootstrap-prep.md:45]: Glad you did this, thanks!  (You can resolve once read)

File: cluster-stamp.bicep
Comment (ckittel) [inline cluster-stamp.bicep:107]: Copy+Paste for all the comments for these newly converted ones, but didn't get them updated to match what they actually are.  Can you do a quick sweep through and update the comments to match the role definition?

**Review by ckittel — APPROVED** [review]
LGTM, thanks!  

Just ONE change to make, with the comments.  ~~But after you make those changes, you can merge~~ (see followup review).  If you want, feel free to copy & paste the description of the role from docs OR you can try to describe it and how it's used in the RI -- either way is fine.

File: cluster-stamp.bicep
Comment (ckittel) [inline cluster-stamp.bicep:46]: Sorry, one more thing .. and this is a blocker.  Can this please be 1.23.5 instead.  1.23.3 is going to be deprecated with the next patch update, and I'd like to not have to chase that.  We could consider, for the first time, going with `1.23` and having a comment in here that says, # For intentionality you'll likely instead wish to pin this to a specific version, such as 1.23.5.

**Review by ckittel — CHANGES_REQUESTED** [review]
One more thing I missed in my last review.

File: cluster-stamp.bicep
Comment (ferantivero) [reply cluster-stamp.bicep:46]: I think we had discussed that once this would be available is what we do wanted to implement. However, I have no personal preference for one or the another.

Im seeing that handling specific versions will require more maintenance effort but at the same time help us to have more control over what the clusters version are being deployed which is better for troubleshooting.

I would let you make the final call in here.

Comment (ferantivero) [reply cluster-stamp.bicep]: in the meantime we are upgrading to `1.23.5` @ckittel 

done | addressed from 73ac8ad3499e0da1c8c5099c92eb20939dab0090

File: cluster-stamp.bicep
Comment (ferantivero) [reply cluster-stamp.bicep]: good catch @ckittel, don't hesitate in proposing wording changes as they are more than welcome. 

done | addressed from e465dbcca392bd8abd68db9bc67ac00c3dbff46b

File: cluster-stamp.bicep
Comment (ckittel) [inline cluster-stamp.bicep:107]:
```suggestion
// Built-in Azure RBAC role that is applied to a cluster to grant its monitoring agent's identity with publishing metrics and push alerts permissions.
```

File: cluster-stamp.bicep
Comment (ckittel) [inline cluster-stamp.bicep:113]:
```suggestion
// Built-in Azure RBAC role that can be applied to an Azure Container Registry to grant the authority pull container images. Granted to the AKS cluster's kubelet identity.
```

File: cluster-stamp.bicep
Comment (ckittel) [inline cluster-stamp.bicep:119]:
```suggestion
// Built-in Azure RBAC role that must be applied to the kublet Managed Identity allowing it to further assign adding managed identities to the cluster's underlying VMSS.
```

File: cluster-stamp.bicep
Comment (ckittel) [inline cluster-stamp.bicep:125]:
```suggestion
// Built-in Azure RBAC role that is applied a Key Vault to grant with metadata, certificates, keys and secrets read privileges.  Granted to App Gateway's managed identity.
```

File: cluster-stamp.bicep
Comment (ckittel) [inline cluster-stamp.bicep:131]:
```suggestion
// Built-in Azure RBAC role that is applied to a Key Vault to grant with secrets content read privileges. Granted to both Key Vault and our workload's identity.
```

**Review by ckittel — COMMENTED** [review]
Some updated comment suggestions.

File: cluster-stamp.bicep
Comment (ckittel) [reply cluster-stamp.bicep]: I'm good with 1.23.5 for now.  Thanks!

File: cluster-stamp.bicep
Comment (ckittel) [reply cluster-stamp.bicep]: Marking this one resolved, but I did add some wording suggestions in a comment below, which included a couple typo fixes.

**Review by ckittel — APPROVED** [review]
🚀

**Pull request merged**
